web/modules/custom/dpl_event/src/Plugin/rest/resource/v1/EventsResource.php

```diff
@@ -243,23 +244,27 @@ public function getPluginDefinition(): array {
    * GET request: Get all eventinstances, hopefully cached.
    */
   public function get(Request $request): Response {
-    // Entity query, pulling all eventinstances.
-    $storage = $this->entityTypeManager->getStorage('eventinstance');
-    $query = $storage->getQuery()
-      ->accessCheck(FALSE)
-      ->condition('status', TRUE)
-      ->sort('date.value');
-
     // Getting a possible from_date URL parameter, and use it in the look-up,
     // to only find events that start from and after this date.
     $typed_request = new RequestTyped($request);
     $from_date = $typed_request->getDateTime('from_date');
 
-    if ($from_date) {
-      $formatted_date = $from_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT);
-      $query->condition('date.value', $formatted_date, '>=');
+    // If no explicit date is set, we'll create a default to only show
+    // upcoming events.
+    if (!$from_date) {
+      $from_date = new DateTime('today');
     }
 
+    $formatted_from_date = $from_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT);
+
+    // Entity query, pulling all eventinstances.
+    $storage = $this->entityTypeManager->getStorage('eventinstance');
+    $query = $storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('status', TRUE)
```

Comment:
This is outside the scope of this PR but I wonder why we disable the access check here. That does not seem right. As I see it we allow anonymous and authenticated access to viewing event instances.